fundingmanager.go

```diff
@@ -2699,21 +2699,17 @@ func (f *fundingManager) handleErrorMsg(fmsg *fundingErrorMsg) {
 	// If this isn't a simple error code, then we'll display the entire
 	// thing.
 	if len(protocolErr.Data) > 1 {
-		resCtx.err <- grpc.Errorf(
+		err = grpc.Errorf(
 			lnErr.ToGrpcCode(), string(protocolErr.Data),
 		)
 	} else {
 		// Otherwise, we'll attempt to display just the error code
 		// itself.
-		resCtx.err <- grpc.Errorf(
+		err = grpc.Errorf(
 			lnErr.ToGrpcCode(), lnErr.String(),
 		)
 	}
-
-	if _, err := f.cancelReservationCtx(peerKey, chanID); err != nil {
-		fndgLog.Warnf("unable to delete reservation: %v", err)
-		return
-	}
+	resCtx.err <- err
```

Comment:
It doesn't seem that this actually solves the issue this PR was meant to. It should be safe to call `handleErrorMsg` in a completely async fashion.

Reply:
It should solve it, as we now cancel the reservation _before_ we send the error on the channel, while before we did it after.

Good point about the async calling though, I will attempt to write a test that exercises that and make sure all necessary mutexes are in place.